lib/streamlit/elements/exception.py

```diff
@@ -205,4 +217,15 @@ def _get_stack_trace_str_list(exception, strip_streamlit_stack_entries=False):
 
     stack_trace_str_list = [item.strip() for item in stack_trace_str_list]
 
+    if isinstance(exception, UncaughtAppException):
+        stacktrace_list = traceback.format_exc().splitlines()
+        stack_trace_str_list = []
+        while stacktrace_list:
+            entry = stacktrace_list.pop()
+            if "Traceback (most recent call last):" in entry:
+                break
+            stack_trace_str_list.insert(0, entry)
+        # remove sensitive information
+        stack_trace_str_list.pop()
```

Comment:
Two things here:

1. It'd be good to add some tests for this behavior to make sure we're parsing stacktraces correctly. I think adding some tests to `st.exception` for this would be ideal
2. Minor given how long stacktraces are in practice, but technically inserting to the front of a list repeatedly like this makes the loop an O(n^2) operation since the whole list needs to repeatedly be shifted over. A better way to write this would be to append to the end of the list, then reverse the list at the end